### PR TITLE
tests: Enable on Windows

### PR DESCRIPTION
**Pull request by andy5995**

### CODE REVIEW

File: src/Lib/package.cpp [hunk 1]
```diff
@@ -109,7 +109,9 @@ void test_datadir(void) {
   return;
 }
 
-int main(void) {
+// main() must be defined with the args in this format, otherwise we may get an
+// "undefined reference to SDL_main"
+int main(int argc, char *argv[]) {
```
Comment (andy5995) [inline src/Lib/package.cpp:114]: This finally fixed the "undefined reference to SDL_main" and the build of the binary and tests complete. Both tests fail though. I haven't investigated yet.

File: tests/meson.build [hunk 2]
```diff
@@ -3,15 +3,24 @@ test_cases = [
   '../src/Lib/Util/FileSystem'
   ]
 
+# Tests fail to compile if the shared library is used on Windows
```
Comment (andy5995) [inline tests/meson.build:6]: @eli-schwartz Do you know why compiling the tests using the netpanzer shared lib fails on Windows but not on the other platforms we're testing on? (macos{amd64,arm64}, Ubuntu, and FreeBSD)

**Comment by andy5995** [comment]
@winrid Whenever you're ready, could you please pull this branch, build, run `meson devenv` and then `meson test --suite=netpanzer -v`

**Comment by andy5995** [comment]
> @winrid Whenever you're ready, could you please pull this branch, build, run `meson devenv` and then `meson test --suite=netpanzer -v`

After you run `meson devenv`, check the value of the env variable NETPANZER_DATADIR and see if it matches with the correct path of where the data is.

File: src/Lib/2D/CachedFontRenderer.cpp [hunk 3]
```diff
@@ -98,3 +106,29 @@ void CachedFontRenderer::cleanup() {
...
   LOGGER.debug("Cached font cleanup: Done.");
 }
+
+#else
+
+#include "test.hpp"
+#include "CachedFontRenderer.hpp"
+
+void CachedFontRenderer::test_openFont(void) {
+  assert (openFont() != NULL);
```
Comment (andy5995) [inline src/Lib/2D/CachedFontRenderer.cpp:116]: This test is failing on my system even though it appears correctly:

```
stderr:
2024-01-26 23:55:25 font path: /home/andy/src/netpanzer/netpanzer/data/fonts/GNUUnifont9FullHintInstrUCSUR.ttf
```

I'm still investigating.

Comment (andy5995) [reply src/Lib/2D/CachedFontRenderer.cpp:117]: Fixed.

File: tests/meson.build (hunk 2 above)
Comment (eli-schwartz) [reply tests/meson.build:7]: Somehow libunwind is being linked in twice?

**Comment by andy5995** [comment]
Rebased

**Comment by andy5995** [comment]
Rebased.

File: tests/meson.build (hunk 2 above)
Comment (andy5995) [reply tests/meson.build]: @eli-schwartz Probably if you look at the improvements I just made in https://github.com/netpanzer/netpanzer/pull/183/commits/ebd7b4b029b3122a31d95bbff0ee7a44d7ccd3c3 you'll see the likely problem. Can this be improved anymore? Do I really need to declare two dependency objects, one for static and one for shared? Is there a bit of a shortcut?

File: meson.build [hunk 4]
```diff
@@ -147,15 +129,23 @@ deps += sdl2_ttf_dep
...
+      has_headers: 'freetype.h',
+      required: false
+      )
     if not freetype2_dep.found()
       freetype2_dep = dependency(
         'freetype2',
         required: true,
         fallback: 'freetype2',
-        default_options: ['default_library=static'])
+        default_options: ['default_library=static', 'png=disabled', 'zlib=disabled'])
```
Comment (andy5995) [inline meson.build:148]: @eli-schwartz Can you tell me why this has absolutely no effect?

```
freetype2 

  Configuration Options Summary:
    OS      : linux
    Zlib    : yes
    Bzip2   : no
    Png     : yes
    Harfbuzz: no
    Brotli  : no

physfs 3.2.0

  Archive support
    zip    : enabled
    7z     : enabled
    grp    : disabled
    wad    : disabled
    hog    : disabled
    mvl    : disabled
    qpak   : disabled
    slb    : disabled
    iso9660: disabled
    vdf    : disabled

netpanzer 0.9.0.07.555

  Subprojects
    freetype2 : YES 8 warnings
    libpng    : YES
    lua       : YES
    ogg       : YES
    physfs    : YES
    sdl2      : YES
    sdl2_mixer: YES
    sdl2_ttf  : YES 1 warnings
    vorbis    : YES
    zlib      : YES
```

Comment (eli-schwartz) [reply meson.build:165]: Most often that happens when you add this after the fact, and your existing build directory has been configured with different settings. The default options only kick in when newly setting up the build directory (they are only defaults, after all).

Comment (andy5995) [reply meson.build:163]: I saw that in the docs, but that isn't the case. I run `meson setup _build` after making the changes. I've done it several times. I'm using meson v1.3.1 installed from pip on Ubuntu Jammy

Comment (andy5995) [reply meson.build:163]: Several times... either after removing the _build directory or using a different name. ;)

**Comment by andy5995** [comment]
@winrid This is now completing successfully. The three tests are running and completing successfully.

File: meson.build (hunk 4 above)
Comment (eli-schwartz) [reply meson.build:163]: Then probably another dependency that is found earlier, is internally using `dependency(freetype2')` and causing it to be configured without any options. Try looking for freetype first.

**Comment by andy5995** [comment]
Rebased

**Comment by winrid** [comment]
Wooooo

**Comment by andy5995** [comment]
This didn't get an extreme amount of testing yet, but it passed all 11 CI tests, as well as me doing some testing locally in a docker container. My point is that there may be some hiccups yet but they should be pretty minor.

**Pull request merged**